From 02b5bd794ecd96d69b1c15db33d06351d5d95bb1 Mon Sep 17 00:00:00 2001
From: Durgadoss R <durgadoss.r@intel.com>
Date: Wed, 30 Sep 2015 18:03:00 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Remove HBR2 from CHV upfront link
 training

CHV does not support HBR2. Hence, this patch removes HBR2 from
the set of values used for upfront link training.

SQUASHME! - This patch should be squashed into the following existing patch:
Author:    Durgadoss R <durgadoss.r@intel.com>
Date:      Fri, 22 May 2015 14:30:07 +0530
Change-Id: I2ae866b32a0998abcf32826efa2504e203dc8ffa
Subject:   FOR_UPSTREAM [VPG]: drm/i915: Enable Upfront link training
           for type-C DP support

Change-Id: I3bee0d9bf4b1f30c85c0a3ccd7af422444f3b3e7
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-4819
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 31b4d99..5595ae9 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -15046,6 +15046,10 @@ start_link_train:
 	intel_dp->lane_count = intel_dp->dpcd[DP_MAX_LANE_COUNT] &
 					DP_MAX_LANE_COUNT_MASK;
 
+	/* CHV does not support HBR2 */
+	if (intel_dp->link_bw == DP_LINK_BW_5_4)
+		intel_dp->link_bw = DP_LINK_BW_2_7;
+
 	do {
 		/* Find port clock from link_bw */
 		crtc->config.port_clock =
@@ -15084,9 +15088,6 @@ start_link_train:
 			intel_dp->lane_count = 2;
 		} else if (intel_dp->lane_count == 2) {
 			intel_dp->lane_count = 1;
-		} else if (intel_dp->link_bw == DP_LINK_BW_5_4) {
-			intel_dp->link_bw = DP_LINK_BW_2_7;
-			intel_dp->lane_count = 4;
 		} else if (intel_dp->link_bw == DP_LINK_BW_2_7) {
 			intel_dp->link_bw = DP_LINK_BW_1_62;
 			intel_dp->lane_count = 4;
-- 
1.9.1

